app.rb

```diff
@@ -0,0 +1,98 @@
+
+module Enumerable
+  def my_each(&prc)
+    length.times do |i|
+      prc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&prc)
+    length.times do |i|
+      prc.call(self[i], i)
+    end
+    self
+  end
+  
+  def my_select(&prc)
+    selected = []
+    my_each do |el|
+      selected << el if prc.call(el) == true
+    end
+    selected
+  end
+  
+  def my_all?(arg = nil)
+    if arg
+      my_each { |ele| return false unless arg === ele }
+    elsif block_given?
+      my_each { |ele| return false unless yield(ele) }
+    else
+      my_each { |ele| return false unless ele }
+    end
+    true
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method, also is not passing all test cases. 

### `all?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are not false.
```
[1, 'string', true].all?
# true
```
- Returns `false` if it finds at least one element that is false or nil
```
[1, 'string', nil].all?
# false
```

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)